build.go

```diff
@@ -160,24 +160,32 @@ func persistStoredFieldValues(fieldID int,
 func InitSegmentBase(mem []byte, memCRC uint32, chunkMode uint32,
 	fieldsMap map[string]uint16, fieldsInv []string, numDocs uint64,
 	storedIndexOffset uint64, fieldsIndexOffset uint64, docValueOffset uint64,
-	dictLocs []uint64) (*SegmentBase, error) {
+	dictLocs []uint64, sectionsIndexOffset uint64) (*SegmentBase, error) {
```

Comment:
I thought fieldsIndex, docValuesIndex and storedValuesIndex will all going to get "sectioned"? In which case - placing a "sectionsIndexOffset" alongside storedIndexOffset, fieldsIndexOffset, docValueOffset seems somewhat misleading?

I gather sectionsIndexOffset will point to that section that holds either the FST or the vector index.

Reply:
i've trimmed down the method signatures, please let me know if they look good.